build.zig

```diff
@@ -0,0 +1,383 @@
+// THIS MATERIAL IS PROVIDED AS IS, WITH ABSOLUTELY NO WARRANTY EXPRESSED
+// OR IMPLIED.  ANY USE IS AT YOUR OWN RISK.
+//
+// Permission is hereby granted to use or copy this program
+// for any purpose, provided the above notices are retained on all copies.
+// Permission to modify the code and to distribute modified code is granted,
+// provided the above notices are retained, and a notice that the code was
+// modified is included with the above copyright notice.
+
+const std = @import("std");
+const Path = std.Build.LazyPath;
+const print = @import("std").debug.print;
+
+pub fn build(b: *std.build.Builder) void {
+    const optimize = b.standardOptimizeOption(.{});
+    const target = b.standardTargetOptions(.{});
+    const t = target.toTarget();
+    var default_enable_threads = true;
+    if (t.isWasm()) { // matches both emscripten and wasi
+        default_enable_threads = false;
+    }
+    const enable_cplusplus = b.option(
+        bool, "enable_cplusplus", "C++ support") orelse false;
+    const build_shared_libs = b.option(
+        bool, "build_shared_libs",
```

Comment:
I think it is better change name to all caps (BUILD_SHARED_LIBS) as in cmake script

Reply:
Fixed!